http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/RuntimeRoutes.scala

```diff
@@ -254,6 +280,56 @@ object RuntimeRoutes {
     )
   }
 
+  implicit val updateGceConfigDecoder: Decoder[UpdateRuntimeConfigRequest.GceConfig] = Decoder.instance { x =>
+    for {
+      machineType <- x.downField("updatedMachineType").as[Option[MachineTypeName]]
+      diskSize <- x
+        .downField("updatedDiskSize")
+        .as[Option[Int]]
+        .flatMap(x => if (x.exists(_ < 0)) Left(negativeNumberDecodingFailure) else Right(x))
+    } yield UpdateRuntimeConfigRequest.GceConfig(machineType, diskSize)
+  }
+
+  implicit val updateDataprocConfigDecoder: Decoder[UpdateRuntimeConfigRequest.DataprocConfig] = Decoder.instance { x =>
+    for {
+      masterMachineType <- x.downField("updatedMasterMachineType").as[Option[MachineTypeName]]
+      diskSize <- x
+        .downField("updatedMasterDiskSize")
+        .as[Option[Int]]
+        .flatMap(x => if (x.exists(_ < 0)) Left(negativeNumberDecodingFailure) else Right(x))
+      numWorkers <- x.downField("updatedNumberOfWorkers").as[Option[Int]].flatMap {
+        case Some(x) if x < 0  => Left(negativeNumberDecodingFailure)
+        case Some(x) if x == 1 => Left(oneWorkerSpecifiedDecodingFailure)
+        case x                 => Right(x)
+      }
+      numPreemptibles <- x.downField("updatedNumberOfPreemptibleWorkers").as[Option[Int]].flatMap {
+        case Some(x) if x < 0 => Left(negativeNumberDecodingFailure)
+        case x                => Right(x)
+      }
+    } yield UpdateRuntimeConfigRequest.DataprocConfig(masterMachineType, diskSize, numWorkers, numPreemptibles)
+  }
+
+  implicit val updateRuntimeConfigRequestDecoder: Decoder[UpdateRuntimeConfigRequest] = Decoder.instance { x =>
+    for {
+      cloudService <- x.downField("cloudService").as[CloudService]
+      r <- cloudService match {
+        case CloudService.Dataproc =>
+          x.as[UpdateRuntimeConfigRequest.DataprocConfig]
+        case CloudService.GCE =>
+          x.as[UpdateRuntimeConfigRequest.GceConfig]
+      }
+    } yield r
+  }
+
+  implicit val updateRuntimeRequestDecoder: Decoder[UpdateRuntimeRequest] = Decoder.instance { x =>
+    for {
+      rc <- x.downField("updatedRuntimeConfig").as[Option[UpdateRuntimeConfigRequest]]
+      as <- x.downField("allowStop").as[Option[Boolean]]
+      ap <- x.downField("updatedAutopause").as[Option[Boolean]]
+      at <- x.downField("updatedAutopauseThreshold").as[Option[Int]]
```

Comment:
use `updatedAutopauseThreshold` to `Option[FiniteDuration]`

Reply:
I left this as `Int` because we interpret it as `minutes` and convert to `FiniteDuration` in the yield block.